### DISCUSSION TITLE
Simplest/Minimum install for a single-node swarm?

### DISCUSSION THREAD

**Discussion by appurist** [BretFisher/ama#234] [Q&A]
First, thank you so much for [the answer](https://github.com/BretFisher/ama/discussions/146#discussioncomment-1207415) to the swarm vs compose question. That also applied in my case. We have a critical but low-load ELK + custom python server and will probably also start the same way, but sometime in the next year add some HA redundancy just to be more safe, after more familiarity with the simpler single-node setup.

What is your recommended installation / method for a simple Swarm node? I have been using the get.docker.com script to get started, but that includes Docker Compose; do you see that as a problem (in terms of avoiding having to keep it up to date, even if not used)?

Or perhaps the simplest is to use the script and then just uninstall Docker Compose?  Or something like that?

I suspect a minimum Swarm node is similar to the script but with Docker Compose uninstalled? If so, should I uninstall just the last two of these or all three? Or can I just ignore their installation as benign?

```
# apt list | grep docker-compose
docker-compose-plugin/noble,now 2.27.0-1~ubuntu.24.04~noble amd64 [installed]
docker-compose-v2/noble 2.24.6+ds1-0ubuntu2 amd64
docker-compose/noble 1.29.2-6ubuntu1 all
```

Actually that was a list of available, and only `docker-compose-plugin` is installed. Safe to leave? Safe to remove?

**Comment by BretFisher** [comment]
It's harmless, and I would leave it be.

Modern (v2) Compose CLI, is just a single binary that acts as a Docker CLI plugin, and doesn't affect Swarm Mode or Docker Engine by just being installed. I would leave the defaults and I've never seen someone bother to uninstall it on servers.  It'll just get upgraded when a `apt upgrade` is done, but it will never get executed unless you run `docker compose`.

**Reply by appurist** [reply]
Perfect, thanks so much for that answer. I came to that conclusion and left it as-is, but it is really good to hear this answer from a subject matter expert like you!
